Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -397,36 +397,38 @@ private class AttributeRemover: SyntaxRewriter {
     var filteredAttributes: [AttributeListSyntax.Element] = []
     for case .attribute(let attribute) in node {
       if attributesToRemove.contains(attribute) {
-        var leadingTrivia = node.leadingTrivia
+        var leadingTrivia = attribute.leadingTrivia
         if let lastNewline = leadingTrivia.pieces.lastIndex(where: { $0.isNewline }),
           leadingTrivia.pieces[lastNewline...].allSatisfy(\.isWhitespace),
-          node.trailingTrivia.isEmpty,
-          node.nextToken(viewMode: .sourceAccurate)?.leadingTrivia.first?.isNewline ?? false
```

Comment:
If we remove this check, then the following test case fails (yes, you don’t need to put spaces between attributes 🤯)

```swift
assertMacroExpansion(
  """
  struct Foo {
    @Test@State var x: Int
  }
  """,
  expandedSource: """
    struct Foo {
      @State var x: Int
    }
    """,
  macros: ["Test": NoOpPeerMacro.self]
)
```

And while writing this, I also realized that we don’t handle `@State@Test` correctly either. Here’s a test case for that

```swift
assertMacroExpansion(
  """
  struct Foo {
    @State@Test
    var x: Int
  }
  """,
  expandedSource: """
    struct Foo {
      @State
      var x: Int
    }
    """,
  macros: ["Test": NoOpPeerMacro.self]
)
```

And a fix suggestion to add this after `triviaToAttachToNextToken += leadingTrivia + trailingTrivia`

```swift
/// If the attribute is not separated by the previous attribute using trivia, eg.
/// `@First@Second var x: Int` (yes, taht's valid Swift), and removing
/// the `@Second` attribute then dropping all trivia of this attribute
/// would join `First` and `var` without any trivia in between, which is
/// invalid. In this case the trailing trivia of the attribute is
/// significant and we need to keep it.
if triviaToAttachToNextToken.isEmpty &&
    attribute.previousToken(viewMode: .sourceAccurate)?.trailingTrivia.isEmpty ?? false {
  triviaToAttachToNextToken = attribute.trailingTrivia
}
```

Reply:
Wow, thanks! I'll add and cover both those test cases.

Reply:
Done.